test/SparqlAntlrParserTest.cpp

```diff
@@ -1647,3 +1650,43 @@ TEST(SparqlParser, updateUnsupported) {
   expectUpdateFails("MOVE DEFAULT TO GRAPH <a>", updateUnsupported);
   expectUpdateFails("COPY GRAPH <a> TO GRAPH <a>", updateUnsupported);
 }
+
+TEST(SparqlParser, UpdateQuery) {
+  auto expectUpdate = ExpectCompleteParse<&Parser::update>{
+      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}}};
+  auto expectUpdateFails = ExpectParseFails<&Parser::query>{};
+  auto Iri = [](std::string_view stringWithBrackets) {
+    return TripleComponent::Iri::fromIriref(stringWithBrackets);
+  };
+  auto Literal = [](std::string s) {
+    return TripleComponent::Literal::fromStringRepresentation(s);
+  };
+
+  expectUpdate("INSERT DATA { <a> <b> <c> }",
+               m::UpdateQuery({}, {{Iri("<a>"), Iri("<b>"), Iri("<c>")}},
+                              m::GraphPattern()));
+  expectUpdate(
+      "INSERT DATA { \"foo:bar\" <b> <c> }",
+      m::UpdateQuery({}, {{Literal("\"foo:bar\""), Iri("<b>"), Iri("<c>")}},
```

Comment:
Does the Grammar allow the subject to be a literal here? And does this have useful semantics?
Because by default, only the object of a triple can be a literal.

Reply:
It parses so its allowed by the grammar. I didn't consider the semantics here.